pkg/position_store/repo.go

```diff
@@ -4,10 +4,8 @@ type PositionValueEncoder func(v interface{}) (string, error)
 type PositionValueDecoder func(s string) (interface{}, error)
 
 type PositionRepo interface {
-	SetEncoderDecoder(encoder PositionValueEncoder, decoder PositionValueDecoder)
-	Get(pipelineName string) (Position, bool, error)
-	GetWithRawValue(pipelineName string) (Position, bool, error)
-	Put(pipelineName string, position Position) error
+	Get(pipelineName string) (*PositionRepoModel, bool, error)
+	Put(pipelineName string, position *PositionRepoModel) error
```

Comment:
why pass pointer of `PositionRepoModel`?  Is the modification inside needed by the caller? 
string itself is already a pointer, pass value is ok I think.

Reply:
string is not passed as pointer, two struct has been defined here
`Position` 

and

`PositionRepoModel`

Reply:
string is not passed as pointer.

Reply:
string is just like a slice of bytes. it stores pointer to a buf and size. 
see `stringStruct` at runtime/string.go

Reply:
@Ryan-Git made several changes:

1. Only keep one `Position` struct, and keep both `Value` and `ValueString`

2. All Get, Put operation using struct (not pointer) as parameter and return type

3. accessors in position_value.go use struct (not pointer) as parameter and return type when we are trying to access internal struct inside `Value`.